### DISCUSSION TITLE
Adding Action to my app

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#28826] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I made a new App ("Foo") with the stork script using recent devel and I'm trying to create an Action called MyAction. It's just a placeholder for now. The app compiles fine when with just the `MyAction` source/header files, but fails when I to register in FooApp.C with `registerSyntax("MyAction", "CustomSyntaxHere");` in the `registerAll` method.

Error:
```
In file included from /home/jesse/projects/moose/framework/build/header_symlinks/MooseApp.h:26,
                 from /home/jesse/projects/foo/build/header_symlinks/FooApp.h:12,
                 from /home/jesse/projects/foo/src/base/FooApp.C:1,
                 from /home/jesse/projects/foo/build/unity_src/base_Unity.C:2:
/home/jesse/projects/foo/src/base/FooApp.C: In static member function 'static void FooApp::registerAll(Factory&, ActionFactory&, Syntax&)':
/home/jesse/projects/moose/framework/build/header_symlinks/ActionFactory.h:26:3: error: invalid use of member function 'Syntax& MooseApp::syntax()' (did you forget the '()' ?)
   26 |   syntax.registerActionSyntax(action, action_syntax, "", __FILE__, __LINE__)
      |   ^~~~~~
/home/jesse/projects/moose/framework/build/header_symlinks/ActionFactory.h:26:3: note: in definition of macro 'registerSyntax'
   26 |   syntax.registerActionSyntax(action, action_syntax, "", __FILE__, __LINE__)
      |   ^~~~~~

```

MyAction.C:
```
#include "MyAction.h"
#include "Factory.h"
#include "Parser.h"
#include "FEProblem.h"

registerMooseAction("FooApp", MyAction, "add_kernel");

InputParameters
MyAction::validParams()
{
  InputParameters params = Action::validParams();

  return params;
}

MyAction::MyAction(const InputParameters & params)
  : Action(params)
{
}

void
MyAction::act()
{
}
```

MyAction.h:
```
#pragma once

#include "Action.h"

class MyAction : public Action
{
public:
  MyAction(const InputParameters & params);

  static InputParameters validParams();

  virtual void act() override;
};
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

The macro registerSyntax is trying to use a Syntax object called syntax 

I think it might be called 's' instead of syntax in this routine

You can add

auto & syntax =s; before the registerSyntax call

**Reply by jessecarterMOOSE** [reply]
That did it. Thanks.... Would this be considered a bug?

**Reply by GiudGiud** [reply]
It s probably just that your app was created when it was s  for the parameter and now it's syntax

**Reply by jessecarterMOOSE** [reply]
No this is a brand new app that I made to test

**Reply by GiudGiud** [reply]
alright then yes it s a bug the macro should work on new apps not just old apps

**Reply by GiudGiud** [reply]
I ll put up a patch
